tests/test_cms_menus.py

```diff
@@ -76,32 +82,46 @@ def test_get_nodes(self):
             attr={"link_target": child2.link_target},
         )
 
-    def test_get_roots_with_multiple_menucontents(self):
-        """Test to check get_roots while creating two menu contents"""
-        menucontent_1 = factories.MenuVersionFactory()
-        menucontent_2 = factories.MenuVersionFactory()
-        roots = self.menu.get_roots(self.request)
-        self.assertEqual(roots.count(), 2)
-        self.assertListEqual(
-            list(roots), [menucontent_1.content.root, menucontent_2.content.root]
-        )
-
-    def test_get_roots_with_versions(self):
+    def test_get_roots_when_draft_mode_isnt_active(self):
         """This test to check versioning would group all the versions
         of menu content and return latest of all distinct menu content
         """
         menucontent_1_v1 = factories.MenuVersionFactory(state=ARCHIVED)
+        # draft version
         menucontent_1_v2 = factories.MenuVersionFactory(
             content__menu=menucontent_1_v1.content.menu
         )
+        # draft version
         menucontent_2_v1 = factories.MenuVersionFactory()
+        # assert to check draft_mode_active is false
+        self.assertFalse(self.menu.renderer.draft_mode_active)
         roots = self.menu.get_roots(self.request)
+
+        # renderer should only render published menucontent
+        # In this testcase there isnt any published version
+        self.assertEqual(roots.count(), 0)
+
+    def test_get_roots_when_draft_mode_active(self):
+        menucontent_1_v1 = factories.MenuVersionFactory(state=ARCHIVED)
+        # draft version
+        menucontent_1_v2 = factories.MenuVersionFactory(
+            content__menu=menucontent_1_v1.content.menu
+        )
+        # draft version
+        menucontent_2_v1 = factories.MenuVersionFactory()
+
+        # Getting renderer to set draft_mode_active
+        renderer = self.renderer
+        renderer.draft_mode_active = True
+        menu = CMSMenu(renderer)
+
+        roots = menu.get_roots(self.request)
         self.assertEqual(roots.count(), 2)
         self.assertListEqual(
             list(roots), [menucontent_1_v2.content.root, menucontent_2_v1.content.root]
         )
 
-    def test_get_roots_with_archived_versions(self):
+    def test_get_roots_for_archived_and_unpublished_versions(self):
```

Comment:
I think you need to test this for both draft mode active and inactive. But these don't have to be a separate tests. You could just add the creation of an unpublished version into the two tests above (and then this test can be deleted).